### PR TITLE
GitHub Copilot Chat with the help of agoncal

### PR DESCRIPTION
**Pull request by agoncal**
#### Check List:
- [x] Tests pass (`./test.sh agoncal` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_agoncal.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [ ] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)

* Execution time: 61.883
* Execution time of reference implementation: 186.944 ms

### CODE REVIEW

File: calculate_average_agoncal.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+# sdk use java 21.0.1-tem
+
+JAVA_OPTS="--enable-preview -XX:+UseShenandoahGC -XX:+UseStringDeduplication -da"
```
Comment (gunnarmorling) [inline calculate_average_agoncal.sh:20]: To use Shenandoah, please add a prepare script which enables an SDK build which provides that GC. You can find an example in _repare_twobiers.sh_ (which uses Temurin).

**Review by gunnarmorling — COMMENTED** [review]
Very cool, welcome to the challenge, @agoncal! It might run a bit longer than the 10 sec am asking for at this point to keep things manageable, but am happy to make an exception for this approach, provided it's valid :)

On that, I'm seeing one difference in the test output:

```
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-10000-unique-keys.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-10.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-1.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-20.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-2.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-3.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-boundaries.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-complex-utf8.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-dot.txt
Validating calculate_average_agoncal.sh -- src/test/resources/samples/measurements-rounding.txt
2c2
< jel;-9.0;17.9;46.5
---
> jel;-9.0;18.0;46.5
```

Note there was a bug in the initial rounding implementation in the baseline, I suppose you are hitting this.

File: calculate_average_agoncal.sh (hunk 1 above)
Comment (agoncal) [reply calculate_average_agoncal.sh:20]: Done with https://github.com/gunnarmorling/1brc/pull/460/commits/10b3e62a268754ba5740d41230125ee0bbe4fb2b

**Comment by agoncal** [comment]
Thanks @gunnarmorling. I arrived late in the game and when I saw the amazing performances of @thomaswue, @royvanrijn and al., I thought "What else could I do?" and I remembered that we were in 2024: therefore, I used AI ;o)

PS: I hadn't synch my repo for a few days and missed the `measurements-rounding` test. It's passing now.

**Comment by gunnarmorling** [comment]
All looking good now. This comes in at 01:09.595, i.e. not exactly the fastest, but super-cool to see the possibilities here. Would love for you to share the experience some more on Twitter or elsewhere :)

**Comment by agoncal** [comment]
Damned. Based on the baseline, I was more expecting 30/40 seconds. Never mind ;o)

Oh, yes. I am planning to write about this experience. I was amazed to see Copilot Chat getting the algorithm right at the first time. The output wasn't good (I needed a few prompts to get the right format), rounding the numbers also wasn't great (again, a few prompts), but at the end I managed to make it work (slowly) in a few prompts only.

I will write a blog about it and let you know.

Thank again @gunnarmorling for what you've done, for all the work you've put into it. Very fun to be part of that. Thank you.

**Comment by gunnarmorling** [comment]
> Damned. Based on the baseline, I was more expecting 30/40 seconds. Never mind ;o)

Ha, yeah. Could always be a difference in terms of ISA, etc.

> I will write a blog about it and let you know.

Excellent, looking forward to it!

> Thank again @gunnarmorling for what you've done, for all the work you've put into it. Very fun to be part of that. Thank you.

It's my pleasure and a true honor to have folks like you particpate! Thank you!

**Pull request merged**